assimilation_code/modules/assimilation/algorithm_info_mod.f90

```diff
@@ -35,8 +38,55 @@ module algorithm_info_mod
 integer, parameter :: BOUNDED_NORMAL_RHF = 101 
 
 public :: obs_error_info, probit_dist_info, obs_inc_info, &
+          init_algorithm_info_mod, end_algorithm_info_mod, &
           EAKF, ENKF, BOUNDED_NORMAL_RHF, UNBOUNDED_RHF, GAMMA_FILTER
 
+!Creates the type definitions for the QCF table
+type obs_error_info_type
+   logical               :: bounded_below, bounded_above
+   real(r8)              :: lower_bound,   upper_bound
+end type
+
+type probit_inflation_type
+   character(len=129)    :: dist_type
+   logical               :: bounded_below, bounded_above
+   real(r8)              :: lower_bound,   upper_bound
+end type
+
+type probit_state_type
+   character(len=129)    :: dist_type
+   logical               :: bounded_below, bounded_above
+   real(r8)              :: lower_bound,   upper_bound
+end type
+
+type probit_extended_state_type
+   character(len=129)    :: dist_type
+   logical               :: bounded_below, bounded_above
+   real(r8)              :: lower_bound,   upper_bound
+end type
+
+type obs_inc_info_type
+   character(len=129)    :: filter_kind
+   logical               :: rectangular_quadrature, gaussian_likelihood_tails
+   logical               :: sort_obs_inc, spread_restoration
+   logical               :: bounded_below, bounded_above
+   real(r8)              :: lower_bound,   upper_bound
+end type
+
+type qcf_table_data_type
+   type(obs_error_info_type)        :: obs_error_info
+   type(probit_inflation_type)      :: probit_inflation
+   type(probit_state_type)          :: probit_state
+   type(probit_extended_state_type) :: probit_extended_state
+   type(obs_inc_info_type)          :: obs_inc_info
+end type
+
+character(len=129), dimension(4) :: header1
+character(len=129), dimension(29) :: header2
+
+character(len=129), allocatable :: qcf_table_row_headers(:)
+type(qcf_table_data_type), allocatable :: qcf_table_data(:)
+
 ! Provides routines that give information about details of algorithms for 
 ! observation error sampling, observation increments, and the transformations
 ! for regression and inflation in probit space. 
```

Comment:
comment for line 93

This is no longer true? People will not be writing their own versions of algorithm_info_mod.f0

```
! For now, it is convenient to have these in a single module since several
! users will be developing their own problem specific versions of these
! subroutines. This will avoid constant merge conflicts as other parts of the
! assimilation code are updated.
````